### PR TITLE
Rewrite readFile and writeFile

### LINKED ISSUE #1989: Rewrite Deno.readFile and Deno.writeFile to use more primitive ops
Should be a simple fix - perhaps using Buffer for readFile.

Sync versions should be done too.

At the end the readFile and writeFile ops should be removed completely from cli/msg.fbs

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #1989

### CODE REVIEW

File: js/read_file.ts [hunk 1]
```diff
@@ -42,5 +44,8 @@ export function readFileSync(filename: string): Uint8Array {
  *       console.log(decoder.decode(data));
  */
 export async function readFile(filename: string): Promise<Uint8Array> {
-  return res(await dispatch.sendAsync(...req(filename)));
+  const file = await open(filename);
+  const contents = await readAll(file);
+  file.close();
+  return contents;
```
Comment (ry) [inline js/read_file.ts]: 👍 

readFileSync should be similarly easy.

**Comment by bartlomieju** [comment]
@ry it looks like `open` and `writeFile` have incompatible APIs... 

For example:
```ts

const options: WriteFileOptions = {
   create: false,
   append: true,
}  // there is no `OpenMode` that would suffice this condition

Deno.writeFile("foo.txt", ..., options);
```

Should I handle that, by doing `stat` beforehand to check if file exists and bail out if it doesn't?

File: js/read_file.ts (hunk 1 above)
Comment (bartlomieju) [reply js/read_file.ts]: Not really, I just realized that both `open` and `read` have only asynchronous... Is there sane way we can utilize promise to behave like blocking op?

Comment (ry) [reply js/read_file.ts]: read should have a synchronous version... 

As for openSync - you'd need to add that. Shouldn't be too hard.

Comment (bartlomieju) [reply js/read_file.ts]: I guess I can just add `openSync` and `readSync`...

Comment (ry) [reply js/read_file.ts]: (oh looks like you're right. You need to add readSync too.)

**Comment by bartlomieju** [comment]
@ry ready for review

File: js/files.ts [hunk 2]
```diff
@@ -32,40 +38,98 @@ export async function open(
...
-export async function read(rid: number, p: Uint8Array): Promise<ReadResult> {
+export function openSync(filename: string, mode: OpenMode = "r"): File {
+  return resOpen(dispatch.sendSync(...reqOpen(filename, mode)));
+}
+
+/** Open a file and return an instance of the `File` object.
+ *
+ *       (async () => {
+ *         const file = await Deno.open("/foo/bar.txt");
+ *       })();
```
Comment (ry) [inline js/files.ts]: Nice examples!

File: js/deno.ts [hunk 3]
```diff
@@ -6,10 +6,12 @@ export { chdir, cwd } from "./dir";
 export {
   File,
   open,
+  openSync,
   stdin,
   stdout,
   stderr,
   read,
+  readSync,
```
Comment (ry) [inline js/deno.ts]: You can export all these other sync functions too. Seems useful for people?
Also what about writeSync?

File: js/buffer.ts [hunk 4]
```diff
@@ -226,6 +236,25 @@ export class Buffer implements Reader, Writer {
       }
     }
   }
+
+  readFromSync(r: SyncReader): number {
```
Comment (ry) [inline js/buffer.ts]: It would be great to have a unit test for this.

File: js/files.ts [hunk 5]
```diff
@@ -93,17 +157,26 @@ export function close(rid: number): void {
...
     return read(this.rid, p);
   }
 
+  readSync(p: Uint8Array): ReadResult {
+    return readSync(this.rid, p);
+  }
+
   seek(offset: number, whence: SeekMode): Promise<void> {
     return seek(this.rid, offset, whence);
   }
```
Comment (kevinkassimo) [inline js/files.ts]: ... and probably adding `seekSync()`? For symmetry with other sync op and for code that doesn't want to be wrapped in async functions

Comment (bartlomieju) [reply js/files.ts]: Yeah, good suggestion :+1:

File: js/files.ts (hunk 2 above)
Comment (bartlomieju) [reply js/files.ts]: Thanks!

**Comment by bartlomieju** [comment]
@ry @kevinkassimo ready for round two

**Pull request merged**